cfgov/mega_menu/tests/test_frontend_conversion.py

```diff
@@ -95,30 +93,113 @@ def make_test_page(cls, title):
+        request = self.make_request('/auto-loans/')
+
         self.menu.refresh_from_db()
 
         # We expect to see two queries here:
         #
         # 1. Wagtail's site root lookup.
         # 2. Single query to retrieve all pages at once.
         with self.assertNumQueries(2):
-            self.do_conversion(self.menu)
+            self.do_conversion(self.menu, request)
 
     def test_conversion_output(self):
-        self.assertEqual(self.do_conversion(self.menu), [
+        request = self.make_request('/auto-loans/')
+
+        self.assertEqual(self.do_conversion(self.menu, request), [
+            {
+                'selected': True,
+                'overview': {
+                    'url': '/consumer-tools/',
+                    'text': 'Consumer Tools',
+                },
+                'nav_groups': [
+                    {
+                        'title': 'Money Topics',
+                        'title_hidden': False,
+                        'nav_items': [
+                            {
+                                'url': '/auto-loans/',
+                                'text': 'Auto Loans',
+                                'selected': True,
+                            },
+                            {
+                                'url': '/bank-accounts/',
+                                'text': 'Wagtail page with other text',
+                            },
+                            {
+                                'url': '/foo/bar/?baz=1',
+                                'text': 'Non-Wagtail page',
+                            },
+                        ],
+                    },
+                ],
+                'featured_items': [
+                    {
+                        'url': '/featured/1/',
+                        'text': 'First featured link',
+                    },
+                ],
+            },
+            {
+                'overview': {
+                    'url': '/about-us/',
+                    'text': 'Alternate Submenu Title',
+                },
+                'featured_items': [
+                    {
+                        'url': '/featured/2/',
+                        'text': 'Second featured link',
+                    },
+                    {
+                        'url': '/featured/3/',
+                        'text': 'Third featured link',
+                    },
+                ],
+                'other_items': [
+                    {
+                        'url': '/financial-well-being/',
+                        'text': 'First other link',
+                        'icon': 'star',
+                    },
+                    {
+                        'url': '/other/2/',
+                        'text': 'Second other link',
+                        'icon': 'mail',
+                    },
+                ],
+            },
+        ])
+
+    def test_conversion_output_selection_ignores_query_string(self):
```

Comment:
What are your thoughts about adding a test for the specific behavior that's added in this PR? I might be wrong, but this test seems more like #6012

Reply:
Thanks, you're totally right. I'll add one.

Reply:
@schbetsy I've added a test for this behavior; would you mind re-reviewing when you get a chance?